import_jobs.py

```
@@ -110,11 +111,20 @@ def get_jobs_from_zipfile(zipfileobject, guid):
     :return: [lxml.eTree, lxml.eTree,...]"""
     logger.debug("Getting current Jobs for guid: %s", guid)
 
+
+    # Get current worker process id, to prevent race conditions.
+    try:
+        p = current_process()
+        process_id =  p.index
+    except:
```

This seems like a bad idea. Silent failures? I suspect we should be more specific about what we expect to fail here.